### PR TITLE
Initial implementation of accesskey controller

### PR DESCRIPTION
**Pull request by chlunde**
This implements the controller, but it has not been wired to a real bitbucket server rest API client yet.

### CODE REVIEW

**Comment by chlunde** [comment]
cc @larhauga

File: apis/accesskey/v1alpha1/types.go [hunk 1]
```diff
@@ -20,36 +20,63 @@ import (
...
 
 	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/provider-bitbucket-server/internal/clients/bitbucket"
 )
 
 // AccessKeyParameters are the configurable fields of a AccessKey.
 type AccessKeyParameters struct {
-	ConfigurableField string `json:"configurableField"`
+	// +immutable
+	ProjectName string `json:"projectName"`
```
Comment (larhauga) [inline apis/accesskey/v1alpha1/types.go]: I think this also should be `projectKey`, as the bitbucket APIs refer to the project key and not the name directly.

File: examples/accesskey/accesskey.yaml [hunk 2]
```diff
@@ -4,6 +4,11 @@ metadata:
   name: example
 spec:
   forProvider:
-    configurableField: test
+    repoName: test
+    projectName: test
```
Comment (larhauga) [inline examples/accesskey/accesskey.yaml]: I think this also should be `projectKey`, as the bitbucket APIs refer to the project key and not the name directly.

File: apis/accesskey/v1alpha1/types.go [hunk 3]
```diff
@@ -20,36 +20,63 @@ import (
...
+
+	// +immutable
+	RepoName string `json:"repoName"`
```
Comment (larhauga) [inline apis/accesskey/v1alpha1/types.go:36]: `RepoName` is referd to as `RepoSlug` in bitbucket API ref and doc (basically a url sanitized value). But it is often just the reponame, so details :)

File: apis/accesskey/v1alpha1/types.go (hunk 1 above)
Comment (chlunde) [reply apis/accesskey/v1alpha1/types.go]: Thanks, fixed :)

File: apis/accesskey/v1alpha1/types.go (hunk 3 above)
Comment (chlunde) [reply apis/accesskey/v1alpha1/types.go:36]: I'm not sure it this would be better for the user, because the UI simply calls it "Name" / "repository name".

**Pull request merged**